sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/_ownership_manager.py

```diff
@@ -23,74 +21,79 @@ class OwnershipManager(object):
 
     """
     def __init__(
-            self, eventhub_client: EventHubClient, consumer_group_name: str, owner_id: str,
-            partition_manager: PartitionManager, ownership_timeout: float
+            self, eventhub_client, consumer_group_name: str, owner_id: str,
+            partition_manager: PartitionManager, ownership_timeout: float,
+            partition_id: str
     ):
         self.cached_parition_ids = []  # type: List[str]
         self.eventhub_client = eventhub_client
+        self.fully_qualified_namespace = eventhub_client._address.hostname  # pylint: disable=protected-access
         self.eventhub_name = eventhub_client.eh_name
         self.consumer_group_name = consumer_group_name
         self.owner_id = owner_id
         self.partition_manager = partition_manager
         self.ownership_timeout = ownership_timeout
+        self.partition_id = partition_id
+        self._initializing = True
 
     async def claim_ownership(self):
         """Claims ownership for this EventProcessor
-        1. Retrieves all partition ids of an event hub from azure event hub service
-        2. Retrieves current ownership list via this EventProcessor's PartitionManager.
-        3. Balances number of ownership. Refer to _balance_ownership() for details.
-        4. Claims the ownership for the balanced number of partitions.
-
-        :return: List[Dict[Any]]
         """
         if not self.cached_parition_ids:
             await self._retrieve_partition_ids()
-        to_claim = await self._balance_ownership(self.cached_parition_ids)
+
+        if self.partition_id is not None:
+            if self.partition_id in self.cached_parition_ids:
+                return [self.partition_id]
+            raise ValueError(
```

Comment:
It seems this exception is not directly actionable by changing an input type from the customers (which is what ValueError is intended for). Then it should be an explicit exception with instructions on cause and recovery

Reply:
This will only happen when the checkpointstore data is corrupt. For instance users have an eventhub with 10 partitions. They delete the eventhub and create another one with the same name and fewer partitions and use the same checkpointstore. This is a corner case. Will talk with cross-lang teams whether the sdk should stop processing and notify users to fix data,  automatically correct the data, or any other strategies.

I raise this exception for debugging. If we expose to users, yes we should have another exception type.

Reply:
Being you are telling me it's not exposed, Not blocking for preview5, solving